feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,755 @@
+			if averageUtilization > cpuHighUtilization {
+				t.Errorf("ERROR: %s %s Type %-20s %-10s - Utilization: %3d%%, exceeding threshold: %3d%%",
+					timestamp, deviceName, cpu, cpuParent, averageUtilization, cpuHighUtilization)
+			} else {
+				t.Logf("INFO: %s %s Type %-20s %-10s - Utilization: %3d%%",
+					timestamp, deviceName, cpu, cpuParent, averageUtilization)
+			}
+		}
+	}
+
+	if len(cardList) > 0 {
+		t.Errorf("ERROR: Didn't find cpu card for checkCards %s", cardList)
+	}
+}
+
+func TestComponentsNoHighMemoryUtilization(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+	description := "Component"
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	cardList := append(controllerCards, lineCards...)
+	for _, component := range cardList {
+		query := gnmi.OC().Component(component).State()
+		timestamp := time.Now().Round(time.Second)
+		componentState := gnmi.Get(t, dut, query)
+		componentType := componentState.GetType()
+		memoryState := componentState.GetMemory()
+		if memoryState == nil {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Memory data not available",
+				timestamp, deviceName, description, component, componentType)
+			continue
+		}
+
+		memoryAvailable := memoryState.GetAvailable()
+		memoryUtilized := memoryState.GetUtilized()
+		memoryUtilization := uint8((memoryUtilized * 100) / (memoryAvailable + memoryUtilized))
+		if memoryUtilization > memoryHighUtilization {
+			t.Errorf("ERROR: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%% (Threshold: %3d%%)",
+				timestamp, deviceName, description, component, componentType, memoryUtilization, memoryHighUtilization)
+		} else {
+			t.Logf("INFO: %s - Device: %s - %s: %-40s - Type: %-20s - Utilization: %3d%%",
+				timestamp, deviceName, description, component, componentType, memoryUtilization)
+		}
+	}
+}
+
+func TestSystemProcessNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	description := "System CPU Process"
+	threshold := uint8(cpuHighUtilization)
+
+	timestamp := time.Now().Round(time.Second)
+	results := gnmi.GetAll(t, dut, query)
+	for _, result := range results {
+		processName := result.GetName()
+		utilization := result.GetCpuUtilization()
+
+		if utilization > threshold {
+			t.Errorf("%s %s ERROR: %s process %-40s utilization: %3d%%, exceeding %3d%%", timestamp, deviceName, description, processName, utilization, threshold)
+		} else {
+			t.Logf("%s %s INFO: %s process %-40s utilization: %3d%%", timestamp, deviceName, description, processName, utilization)
+		}
+	}
+}
+
+func TestSystemProcessNoHighMemorySpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	query := gnmi.OC().System().ProcessAny().State()
+	description := "System Process Memory"
```

Comment:
can be const

Reply:
Please see response to previous comment related to this. Thanks!

Reply:
Thanks for your comments. Updated code as below:
```go
	query := gnmi.OC().System().ProcessAny().State()
	const description = "System Process Memory"
```